### PR TITLE
Add CARMA Platform System ROS1/ROS2 Status Documentation

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

The README has been updated to include a description of the current breakdown/overview of ROS 1 Noetic and ROS 2 Foxy content existing within the CARMA Platform System. This documentation is being included to help new developers avoid confusion when inspecting the CARMA Platform system's codebase and stumbling across a small amount of ROS 1 Noetic packages in different repositories.

This documentation will be updated as ROS 1 Noetic packages continue to be ported to ROS 2 Foxy, and again when packages are upgraded from ROS 2 Foxy to ROS 2 Humble.

## Related Issue
[ARC-147](https://usdot-carma.atlassian.net/browse/ARC-147) Jira ticket.

## Motivation and Context
See above.

## How Has This Been Tested?
NA

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[ARC-147]: https://usdot-carma.atlassian.net/browse/ARC-147?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,35 +5,50 @@
...
+
+With regards to the primary [carma-platform](https://github.com/usdot-fhwa-stol/carma-platform) repository, the following ROS 1 Noetic content still exists:
+- The `rosbag_mock_drivers` and `mock_lightbar_driver` packages, which serve as mock hardware drivers to support local integration testing. These are expected to be upgraded to ROS 2 Foxy in the near future.
+- The `carma_record` package, which enables ROS 1 rosbags to be recorded. This package is being preserved to enable legacy data analysis scripts to be used for system evaluation.
+- The ROS 1 rviz configuration in the `carma` package, which enables rviz to be run natively when CARMA Platform is deployed on a PC with a ROS 1 Noetic installed to its desktop. This is planned be removed after all ROS 2 Foxy packages are upgraded to ROS 2 Humble.
```
Comment (saina-ramyar) [inline README.md:25]: I thought we already ported in rviz2. Is this statement still valid?

File: README.md [hunk 2]
```diff
@@ -5,35 +5,50 @@
...
+
+Additionally, the [carma-ssc-interface-wrapper](https://github.com/usdot-fhwa-stol/carma-ssc-interface-wrapper) repository contains the drive-by-wire driver and SSC packages that are used for CARMA Platform-supported vehicle classes. The packages for several vehicle classes (Chrysler Pacifica, Ford Fusion, and Freightliner Cascadia) are only available for ROS 1 Noetic. These packages are closed source, and the external company providing them does not plan to upgrade them to ROS 2.
+
+Finally, the [carma-carla-integration](https://github.com/usdot-fhwa-stol/carma-carla-integration) and [carma-ns3-adapter](https://github.com/usdot-fhwa-stol/carma-ns3-adapter) repositories contain packages that are required for XiL deployments. These packages are all built for ROS 1 Noetic due to a dependency on on CARLA 0.9.10, which only exposes a networking bridge for ROS 1. In the future, when XiL-related repositories are upgraded to support a newer version of CARLA, these packages will be upgraded to ROS 2.
```
Comment (saina-ramyar) [inline README.md:29]: Are these readmes for internal use or external? if its expected to be used by external users, maybe change "XiL deployment" to simulation or something more clear

File: README.md (hunk 1 above)
Comment (JonSmet) [reply README.md]: We've created an rviz2 configuration to use, but we are still simultaneously maintaining support for the old ROS 1 rviz configuration until we update all vehicle PCs and VMs to have a desktop installation of ROS 2. I have updated the text to clarify that we are currently supporting both ROS 1 rviz and ROS 2 rviz2.

File: README.md (hunk 2 above)
Comment (JonSmet) [reply README.md]: That's a good catch. I've updated the language to replace 'XiL' references with 'simulation'.

File: README.md [hunk 3]
```diff
@@ -8,7 +8,7 @@
...
+Folders containing the name of a specific vehicle class (such as 'lexus_rx_450h_2019') contain vehicle configuration data that is specific to that class of vehicle and sensor suite (for example, the launch file for the sensors used on that vehicle). Additionally, folders containing the name of a specific simulation scenario contain configuration data that is specific to that specific simulation scenario. These folders also contain the Docker Compose files that will launch CARMA Platform with the appropriate configuration. These folders do not contain calibration information that is specific to an individual vehicle.
```
Comment (saina-ramyar) [inline README.md:11]: In this sentence, second specific is not needed:
"specific to that specific simulation scenario." to "specific to that simulation scenario."

Comment (JonSmet) [reply README.md]: Thanks for catching that, just fixed this.

**Pull request merged**